### DISCUSSION TITLE
docs/permission/permission-configuration

### DISCUSSION THREAD

**Discussion by giscus** [casdoor/casdoor#1946] [Docs comments]
# docs/permission/permission-configuration

Using exposed Casbin APIs to manage users' access rights in organization

https://casdoor.org/docs/permission/permission-configuration

**Comment by thiennc1107** [comment]
Hi, currently the resources field cannot be added as string in the web UI, I'm I missing anything or is this an issue?

**Reply by hsluoyz** [reply]
Fixed in: https://github.com/casdoor/casdoor/commit/05703720c5f2963413d498b70ee373adc560d129

Use `Custom` resource type to enable free input in resources

**Comment by MyColorfulDays** [comment]
I filled a table name in `adapter` field, but the database does not create it.

**Reply by hsluoyz** [reply]
@MyColorfulDays 

1. Did you enable `Same with DB`?
2. Provide your `dataSourceName` in app.conf (hide your credendials)
3. Does your DB user in `dataSourceName` have permission to create table in your DB?

**Reply by MyColorfulDays** [reply]
@hsluoyz 
1. I don't know what is `Same with DB` config, and I am not config it.
2. My `dataSourceName` is `dataSourceName = "user=postgres password=xxx host=172.0.0.1 port=5432 sslmode=disable dbname=casdoor"`
3. The database user is postgresql superuser, and has permission to create table.

**Reply by hsluoyz** [reply]
@MyColorfulDays

1. I mean this switch, you should turn it OFF: https://door.casdoor.com/adapters/built-in/user-adapter-built-in

![image](https://github.com/casdoor/casdoor/assets/3787410/af1f89f7-6ebb-41d8-9fe3-52e42338ee92)

2. What's your postgresql DB version?

**Reply by MyColorfulDays** [reply]
@hsluoyz 
1. The `Use same DB` is `true`, same as your shortcut, the switch is not editable, how to turn it off?
![image](https://github.com/casdoor/casdoor/assets/21351745/c8dd6bae-cfba-4899-b7a5-b3e973c8cdf7)

2. My postgresql version is 15.3

**Reply by MyColorfulDays** [reply]
@hsluoyz 
I turn off it in database and create a new permission, but the table, filled in `adapter`, still not created.
